### PR TITLE
[decoupled-execution] Test Fixes

### PR DESCRIPTION
**Pull request by yuxiamit**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update or suggest changes to the docs at https://developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: config/src/config/consensus_config.rs [hunk 1]
```diff
@@ -47,8 +47,8 @@ impl Default for ConsensusConfig {
             safety_rules: SafetyRulesConfig::default(),
             sync_only: false,
             mempool_poll_count: 1,
-            decoupled_execution: false, // by default, we turn of the decoupling execution feature
-            channel_size: 30,           // hard-coded
+            decoupled_execution: true, // by default, we turn of the decoupling execution feature
```
Comment (zekun000) [inline config/src/config/consensus_config.rs:50]: remove the switch from the PR

File: config/src/config/consensus_config.rs [hunk 2]
```diff
@@ -47,8 +47,8 @@ impl Default for ConsensusConfig {
...
+            channel_size: 60,          // hard-coded
```
Comment (zekun000) [inline config/src/config/consensus_config.rs:51]: why this is necessary?

File: consensus/consensus-types/src/executed_block.rs [hunk 3]
```diff
@@ -13,12 +13,13 @@ use diem_types::{
...
 };
 use executor_types::StateComputeResult;
+use serde::{Deserialize, Serialize};
 use std::fmt::{Debug, Display, Formatter};
 
 /// ExecutedBlocks are managed in a speculative tree, the committed blocks form a chain. Besides
 /// block data, each executed block also has other derived meta data which could be regenerated from
 /// blocks.
-#[derive(Clone, Eq, PartialEq)]
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize)]
```
Comment (zekun000) [inline consensus/consensus-types/src/executed_block.rs:22]: where do we need to serialize it?

File: consensus/consensus-types/src/sync_info.rs [hunk 4]
```diff
@@ -191,9 +191,9 @@ impl SyncInfo {
     }
 
     pub fn has_newer_certificates(&self, other: &SyncInfo) -> bool {
-        self.highest_certified_round() > other.highest_certified_round()
+        self.highest_ledger_info_round() > other.highest_ledger_info_round()
```
Comment (zekun000) [inline consensus/consensus-types/src/sync_info.rs:194]: why the order matter?

File: consensus/safety-rules/src/local_client.rs [hunk 5]
```diff
@@ -77,10 +75,10 @@ impl TSafetyRules for LocalClient {
     fn sign_commit_vote(
         &mut self,
         ledger_info: LedgerInfoWithSignatures,
-        new_ledger_info: LedgerInfo,
+        last_block: ExecutedBlock,
```
Comment (zekun000) [inline consensus/safety-rules/src/local_client.rs:78]: let's skip the extension check for this pr

File: consensus/safety-rules/src/safety_rules.rs [hunk 6]
```diff
@@ -109,6 +110,14 @@ impl SafetyRules {
 
         if self.decoupled_execution {
             Ok(vote_proposal.vote_data_ordering_only())
+            /*
+            // TODO: uncomment after we turn on extension check
+            let new_tree = vote_proposal
+                .accumulator_extension_proof()
+                .next_without_verify()
+                .map_err(|e| Error::InvalidAccumulatorExtension(e.to_string()))?;
+            Ok(vote_proposal.vote_data_with_extension_proof(&new_tree.0))
```
Comment (zekun000) [inline consensus/safety-rules/src/safety_rules.rs:119]: even we do extension check it should not be here?

**Review by zekun000 — COMMENTED** [review]
let's skip the extension check from this one, also I had a few comments on the previous PR

File: config/src/config/consensus_config.rs (hunk 1 above)
Comment (yuxiamit) [reply config/src/config/consensus_config.rs:50]: I'll do this after we pass the cluster test

File: config/src/config/consensus_config.rs (hunk 2 above)
Comment (yuxiamit) [reply config/src/config/consensus_config.rs:51]: The change of channel_size is not necessary. I'll change it back later. This was to test the on-chain reconfiguration (in a later pr).

File: consensus/consensus-types/src/executed_block.rs (hunk 3 above)
Comment (yuxiamit) [reply consensus/consensus-types/src/executed_block.rs:22]: In safety-rule, to perform the extension check (commented), we need to give the safety-rule the executed block (or both the block info and state compute result).

File: consensus/consensus-types/src/sync_info.rs (hunk 4 above)
Comment (yuxiamit) [reply consensus/consensus-types/src/sync_info.rs:194]: consider a node with a qc of round 200, but it has slow execution at HLI 10, and it was set to be sync_only. So this blocks the sync_up path until the local qc is smaller than remote qc. However, during sync_only, the node can still update the local qc through the process_vote path?

File: consensus/safety-rules/src/local_client.rs (hunk 5 above)
Comment (yuxiamit) [reply consensus/safety-rules/src/local_client.rs:78]: OK, I'll change it later.

File: consensus/safety-rules/src/safety_rules.rs (hunk 6 above)
Comment (yuxiamit) [reply consensus/safety-rules/src/safety_rules.rs:119]: we can put it here if we are doing a batch to batch extension check.

File: consensus/consensus-types/src/sync_info.rs (hunk 4 above)
Comment (zekun000) [reply consensus/consensus-types/src/sync_info.rs:194]: worth a comment about the ordering

Comment (yuxiamit) [reply consensus/consensus-types/src/sync_info.rs:194]: added

File: config/src/config/consensus_config.rs (hunk 1 above)
Comment (yuxiamit) [reply config/src/config/consensus_config.rs:52]: changed back

File: config/src/config/consensus_config.rs (hunk 2 above)
Comment (yuxiamit) [reply config/src/config/consensus_config.rs:53]: resolved

File: consensus/consensus-types/src/sync_info.rs (hunk 4 above)
Comment (yuxiamit) [reply consensus/consensus-types/src/sync_info.rs:195]: > consider a node with a qc of round 200, but it has slow execution at HLI 10, and it was set to be sync_only. So this blocks the sync_up path until the local qc is smaller than remote qc. However, during sync_only, the node can still update the local qc through the process_vote path?

actually this is wrong. I was thinking of comparisons...

**Pull request closed without merging**